### PR TITLE
init fwang1

### PR DESCRIPTION
**Pull request by lionelfeng**

### CODE REVIEW

**Comment by lionelfeng** [comment]
cc @updownlife

File: fwang1/solutions/P007.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        if x ==  - MAX_INT - 1: return 0
+        
+        neg_flag = -1 if x < 0 else 1
+        x = -x if x < 0 else x
+
+        res = 0
+        while x:
+            if res > MAX_INT/ (10): return 0
+            digit = x % 10
+            if res * 10 > MAX_INT - digit:
```
Comment (updownlife) [inline fwang1/solutions/P007.py]: For 32-bit int, these checking is not necessary. 
    if res > MAX_INT/10: return 0
    else:  res = res * 10 + x%10
Assume res < MAX_INT/10, then res * 10 + x%10 will not overflow.
Assume res ==MAX_INT/10== 2147483647/10==214748364
then reverse(x) = 214748364m, x=m463847412. As x <= 2147483647(x must be a valid int as input), so m<=1. Hence, reverse(x)=2147483640 or 2147483641, which are both less than INT_MAX

Comment (updownlife) [inline fwang1/solutions/P007.py]: But generally, you are right to add these checking for extending it to other length of integers

File: fwang1/solutions/P094.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+class Solution(object):
+    def inorderTraversal(self, root):
+        """
+        :type root: TreeNode
+        :rtype: List[int]
+        """
+        if not root: return []
+        stack = [root]
+        result = []
+        while len(stack) != 0:
```
Comment (updownlife) [inline fwang1/solutions/P094.py]: use "while stack:" is more compact

File: fwang1/solutions/P367.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import math
+class Solution(object):
+    def isPerfectSquare(self, num):
+        start = int(math.pow(2, math.floor(math.log(num, 2)/ 2)))
```
Comment (updownlife) [inline fwang1/solutions/P367.py]: This is smart!

File: fwang1/solutions/P367.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        end =  start * 2
```
Comment (updownlife) [inline fwang1/solutions/P367.py]: Why "end =  start * 2"?

Comment (lionelfeng) [inline fwang1/solutions/P367.py]: do you have a smaller range based on the log2?

File: fwang1/solutions/P094.py (hunk 2 above)
Comment (lionelfeng) [inline fwang1/solutions/P094.py]: right! I saw that in your shangqiang-ed PR

File: fwang1/solutions/P367.py (hunk 4 above)
Comment (updownlife) [inline fwang1/solutions/P367.py]: no, it seems start * 2 is the lower bound

**Comment by updownlife** [comment]
LGTM

**Pull request merged**